### PR TITLE
feat: room package part 1

### PR DESCRIPTION
**Pull request by faiq-naufal**
> [!NOTE]
> This PR should be reviewed and squash merged first to reduce the PR complexity of the [complete version](https://github.com/inlivedev/inlive-js-sdk/pull/66) of the initial room package implementation. Some missing modules should be available in the complete version

# Description
This is the first part of the `room` package implementation. It mainly covers:
- The complete README and references docs for `room` package. You can [read the README here](https://github.com/inlivedev/inlive-js-sdk/blob/feat/room-package-part-1/packages/room/README.md)
- The entrypoint for the `room` package
- The API module implementation which integrates to the `/room/` Hub API endpoints.
- The facade module that provides a unified interface for all modules implemented for this package.
- Generating bundle and type declaration file for the `room` package

### CODE REVIEW

File: packages/room/README.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+### Room object
+
+The room object is the object created when the `Room` module is initialized. It consists methods that relates to the room scope. Some methods that directly interact to the room API can run on both client and server sides.
+
+#### Sample Code
+
+```js
+// create a new room
+const newRoom = await room.createRoom('a new room');
```
Comment (tyohan) [inline packages/room/README.md:25]: @faiq-naufal is this support to create room with a custom ID?

**Review by tyohan — COMMENTED** [review]
@faiq-naufal I only have one question on review

File: packages/room/README.md (hunk 1 above)
Comment (faiq-naufal) [reply packages/room/README.md]: Mas @tyohan , this is already updated with this commit [3809023](https://github.com/inlivedev/inlive-js-sdk/pull/68/commits/3809023d41d4265abf4d721cec22f5ba2dadb294)

**Review by tyohan — APPROVED** [review]
LGTM

**Pull request merged**